MotionPhotoReader/app/src/main/java/com/example/motionphotoreader/MotionPhotoInfo.java

```diff
@@ -0,0 +1,95 @@
+package com.example.motionphotoreader;
+
+import android.media.MediaExtractor;
+import android.media.MediaFormat;
+import android.media.MediaMetadataRetriever;
+import android.net.Uri;
+import android.util.Log;
+
+import androidx.annotation.VisibleForTesting;
+
+import com.adobe.internal.xmp.XMPException;
+import com.adobe.internal.xmp.XMPMeta;
+
+import java.io.File;
+import java.io.FileDescriptor;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/**
+ * Holds data about the Motion Photo file.
+ */
+public class MotionPhotoInfo {
+
+    private int width;
+    private int height;
+    private long duration;
+
+    private int videoOffset;
+    private long presentationTimestampUs;
```

Comment:
These shouldn't change once they are set, so consider marking these `final` having these passed to the private constructor.